src/Modifiers/CoreModifiers.php

```diff
@@ -1227,7 +1228,10 @@ public function mailto($value, $params)
      */
     public function markdown($value, $params)
     {
-        if (! is_string($value)) {
+        if ($value instanceof ArrayableString) {
+            // NOTE: Could use 'instanceof Stringable' instead (and then cast to string) in PHP8+
+            $value = $value->value();
+        } elseif (! is_string($value)) {
             return $value;
         }
```

Comment:
I think we might be able to just cast $value to a string rather than using these conditions.

Reply:
Would be nice. In my tests I get an error though: 'Object of class ... could not be converted to string' when casting an object to string that doesn't implement `__toString()`.

Reply:
Which test?

Reply:
Quick and dirty, I put something like this in phptester.net, the last statement will fail:

```php
<?php


class Willing
{
	public function __toString()
	{
		return 'okay!';
	}
}


class Unwilling
{
}


$value = new Willing;

$value = (string) $value;


$value = new Unwilling;

$value = (string) $value;


```

Reply:
It doesn't make sense for someone to apply a markdown modifier to something that doesn't evaluate to a string.

Reply:
Yeah I agree. You think it should just fail loud then?

Reply:
I'm guessing you do 🎲